src/stream/src/task/stream_manager.rs

```diff
@@ -573,13 +577,31 @@ impl LocalStreamManagerCore {
                     *ENABLE_ASYNC_STACK_TRACE,
                 );
                 let instrumented = monitor.instrument(traced);
-                tokio::spawn(instrumented)
+
+                let memory_usage_metrics = async move {
+                    loop {
+                        TASK_LOCAL_ALLOC.with(|alloc| {
+                            metrics
+                                .actor_memory_usage
+                                .with_label_values(&[&actor_id_str])
+                                .set(alloc.bytes_in_use() as i64)
+                        });
+
+                        tokio::time::sleep(Duration::from_secs(1)).await;
+                    }
+                };
+
+                let allocated =
+                    TASK_LOCAL_ALLOC.scope(LocalStatsAlloc::new(std::alloc::Global), async move {
+                        futures::join!(instrumented, memory_usage_metrics);
```

Comment:
Will memory_usage_metrics's loop infinitely block join from finishing?

Reply:
+1. I think we should select here?